### PR TITLE
[15.0][MIG] stock_inventory_discrepancy: Migration to version 15.0

### PR DESCRIPTION
**Pull request by ernestotejeda**
Cc @Tecnativa TT36993

I made this 'PR' apart from https://github.com/OCA/stock-logistics-warehouse/pull/1573 because it has a different approach. I have based more on the native odoo 15 approach to do the inventory adjustment and not make the functionalities of this module dependent on the new stock_inventory OCA module.

Main changes:
- The access group that gives rights to make inventory adjustments is now independent of the stock 'user' group
- Discrepancy threshold fields are now displayed on stock.quant tree view and highlighted when the threshold is excceded. 
- Applying an inventory adjustment (adjustment in stock.quant) both from an inventory line and from the button in the header of the stock.quant tree view, will launch an alert wizard showing the quants with discrepancy and only users with the access group 'Validate Inventory Adjustments exceeded threshold' can proceed to apply the adjustment from this wizard. 
- Adapt tests.

ping @carlosdauden @sergio-teruel @pedrobaeza

### CODE REVIEW

File: stock_inventory_discrepancy/wizards/confirm_discrepancy_wiz.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="stock_quant_confirm_discrepancy" model="ir.ui.view">
+        <field name="model">stock.quant</field>
+        <field name="inherit_id" ref="stock.view_stock_quant_tree_inventory_editable" />
+        <field name="mode">primary</field>
+        <field name="arch" type="xml">
+            <button name="action_apply_inventory" position="attributes" />
+            <button name="action_set_inventory_quantity_to_zero" position="replace" />
```
Comment (carlosdauden) [inline stock_inventory_discrepancy/wizards/confirm_discrepancy_wiz.xml]: Using replace you must add priority > 98 in this view

**Review by carlosdauden — CHANGES_REQUESTED** [review]

File: stock_inventory_discrepancy/wizards/confirm_discrepancy_wiz.xml (hunk 1 above)
Comment (rousseldenis) [reply stock_inventory_discrepancy/wizards/confirm_discrepancy_wiz.xml]: I'd prefer setting it as invisible.

**Comment by ernestotejeda** [comment]
@carlosdauden @rousseldenis changes done

**Review by sergio-teruel — APPROVED** [review]
Functional and code review!

Great job!

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**